server/cluster/unsafe_recovery_controller.go

```diff
@@ -509,10 +528,10 @@ func (u *unsafeRecoveryController) changeStage(stage unsafeRecoveryStage) {
 	u.output = append(u.output, output)
 	data, err := json.Marshal(output)
 	if err != nil {
-		u.err = err
-		return
+		log.Error("Unsafe recovery fail to marshal json object", zap.Error(err))
```

Comment:
Do we need to return here?

Reply:
I don't think we need to return here. This is a json parsing error of logging, does not affect the main logic.